src/ott/tools/plot.py

```diff
@@ -100,13 +104,46 @@ def _scatter(self, ot: Transport):
 
   def _mapping(self, x: jnp.ndarray, y: jnp.ndarray, matrix: jnp.ndarray):
     """Compute the lines representing the mapping between the 2 point clouds."""
+    # Only plot the lines with a cost above the threshold.
     u, v = jnp.where(matrix > self._threshold)
     c = matrix[jnp.where(matrix > self._threshold)]
     xy = jnp.concatenate([x[u], y[v]], axis=-1)
+
+    # Check if we want to adjust transparency.
+    adjust_transparency = self._adjust_transparency
+
+    # We can only adjust transparency if max(c) > min(c).
+    if adjust_transparency:
+      min_matrix, max_matrix = jnp.min(c), jnp.max(c)
+      if max_matrix == min_matrix:
+        adjust_transparency = False
+
+    # Initialize the list of lines of the form (start, end, strength, alpha).
     result = []
+
+    # Loop over the lines and add them to the list.
     for i in range(xy.shape[0]):
+
+      # Line strength is coupling strength times the number of points.
```

Comment:
This comment doesn't provide any additional info to the line below, would again remove.